lib/rubocop/cop/layout/space_inside_hash_literal_braces.rb

```diff
@@ -72,29 +72,15 @@ class SpaceInsideHashLiteralBraces < Base
         MSG = 'Space inside %<problem>s.'
 
         def on_hash(node)
-          tokens = processed_source.tokens
+          tokens = processed_source.tokens_within(node)
+          return unless tokens.first.left_brace? && tokens.last.right_curly_brace?
 
-          hash_literal_with_braces(node) do |begin_index, end_index|
-            check(tokens[begin_index], tokens[begin_index + 1])
-            return if begin_index == end_index - 1
-
-            check(tokens[end_index - 1], tokens[end_index])
-          end
+          check(tokens[0], tokens[1])
+          check(tokens[-2], tokens[-1]) if tokens.size > 1
```

Comment:
Previous condition translates to `if tokens.size > 2`, no? (technically `!= 2` but can't be < 2, right?). Still I don't see how that can create an issue, it's just an optimization...

Reply:
Changed to `> 2`, but the same offense.
Funny, that I tested on ruby 2.7 on mac and saw 1 offense on one of the test files, but on CI (non windows) no offenses.

Reply:
Ok, finally, I have figured it out - https://github.com/rubocop-hq/rubocop-ast/pull/116

The problem is that sorting is not stable (and this sometimes  can change the relative order of tokens like `@text=""  @pos=123...123` and `@text="}" @pos=123..124`) and this can lead to incorrect tokens returned by `ProcessedSource#tokens_within`.